dwave/cloud/cli.py

```diff
@@ -896,3 +943,156 @@ def setup(install_all, ask_full, verbose):
+
+    # refresh
+    flow.refresh_token()
+
+    output('Access and refresh tokens successfully refreshed. '
+           'You can now use "dwave auth get" to view them.')
+
+
+@cli.group()
+def leap():
+    """Interact with Leap API."""
+
+
+@leap.group()
+def project():
+    """Leap projects."""
+
+
+@project.command(name='ls')
+@config_file_options()
+@json_output
+@standardized_output
+def list_leap_projects(*, config_file, profile, json_output, output):
+    """List available Leap projects."""
+
+    config = validate_config_v1(load_config(config_file=config_file, profile=profile))
+    flow = LeapAuthFlow.from_config_model(config)
+
+    if not flow.token or 'access_token' not in flow.token:
+        raise CLIError('Leap API access token not found. Please run "dwave auth login".', code=100)
+
+    if flow.token_expires_soon(within=60):
+        output("Access token expired (or expires soon), refreshing it.")
+        flow.refresh_token()
+
+    account = api.LeapAccount.from_config(config=flow.config,
+                                          token=flow.token.get('access_token'))
+    projects = account.list_projects()
+    output("Leap projects:", projects=[p.model_dump() for p in projects])
+    for project in projects:
+        output(f" - {project.name} (code={project.code!r}, id={project.id})")
+
+
+@project.command(name='token')
+@config_file_options()
+@click.option('--project', 'project_slug', required=True, type=str,
+              help='Leap project ID, name or code.')
+@json_output
+@standardized_output
+def leap_project_token(*, config_file, profile, project_slug, json_output, output):
+    """Get Solver API token for a selected Leap project."""
+
+    config = validate_config_v1(load_config(config_file=config_file, profile=profile))
+    flow = LeapAuthFlow.from_config_model(config)
+
+    if not flow.token or 'access_token' not in flow.token:
+        raise CLIError('Leap API access token not found. Please run "dwave auth login".', code=100)
+
+    if flow.token_expires_soon(within=120):
+        output("Access token expired (or expires soon), refreshing it.")
+        flow.refresh_token()
+
+    account = api.LeapAccount.from_config(config=flow.config,
+                                          token=flow.token.get('access_token'))
+    projects = account.list_projects()
+
+    # find project id
+    needle = str(project_slug).strip().lower()
+    project = [p for p in projects if needle in (str(p.id), p.name.lower(), p.code.lower())]
+
+    if not project:
+        raise CLIError(f'Project with {project_slug!r} ID, name or code not found. '
+                       'Please run "dwave leap project ls" to list available projects.', code=101)
+
+    # get token
+    project = project[0]
+    token = account.get_project_token(project=project)
+
+    output(f"Solver API token for project {project.name} ({project.code}) is {token}.",
+           token=token)
```

Comment:
I get an error for one particular project which is (1) last in the list (2) contains another project name but different case:

```
dwave leap project ls --profile test
Leap projects:
 - ... 
 - jp_test2 ...
 - tonytest  ...
 - TonyTestbeta ...
```

`dwave leap project token --project "tonytest" --profile=test` works but

```
dwave leap project token --project "TonyTestbeta" --profile=test
Unhandled error: 1 validation error for _LeapProjectTokenResponse
data.token
  Input should be a valid string [type=string_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.4/v/string_type
```

Reply:
This is a validation error of the Leap API response -- the token for that project is `None`, which according to specs is incorrect. Now, if `None` is in fact allowed value for token returned from Leap API, I'll update expected response specs.

/cc @charleswhchan

Reply:
Charles says this is a corner case for very old projects... I'll extend the model to support `null` tokens.

Reply:
@JoelPasvolsky, this should now be fixed, feel free to test it again.